src/js/leaflet.storage.js

```diff
@@ -683,20 +700,33 @@ L.Storage.Map.include({
                 layer;
             if (layerId) layer = map.datalayers[layerId];
             if (fileInput.files.length) {
-                if (layer) {
-                    layer.importFromFiles(fileInput.files, type);
-                } else {
-                    for (var i = 0, f; f = fileInput.files[i]; i++) {
-                        layer = this.createDataLayer({name: f.name});
-                        layer.importFromFile(f, type);
+                var file;
```

Comment:
I'm wondering if we should not move the for block to a separate function to gain readability. This can be done in a later move though :)